apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx

```diff
@@ -84,83 +70,64 @@ class MultipleChoiceOverviewTable extends Component {
   };
 
-  getColumns = (sortable) => {
-    let dataColumns = [
-      {
-        property: 'question',
-        header: {
-          label: commonMsg.question(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-          }},
-          transforms: [sortable],
-        },
-        cell: {
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-          }}
-        }
-      },
-      {
-        property: 'percentAnsweredOptionOne',
-        header: {
-          label: commonMsg.answerOptionA(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-            width: 90,
-          }},
-        },
-        cell: {
-          format: answerOptionsOneFormatter,
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-            width: 90,
-          }}
-        }
+  getNotAnsweredColumn = () => (
+    {
+      header: {
+        label: commonMsg.notAnswered(),
+        props: {style: tableLayoutStyles.headerCell},
       },
-      {
-        property: 'percentAnsweredOptionTwo',
-        header: {
-          label: commonMsg.answerOptionB(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-            width: 90,
-          }},
-        },
-        cell: {
-          format: answerOptionsTwoFormatter,
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-            width: 90,
-          }}
-        }
+      cell: {
+        props: {style: tableLayoutStyles.cell},
+      }
+    }
+  );
+
+  getAnswerColumn = (index) => (
+    {
+      header: {
+        label: alphabetMapper[index],
+        props: {style: tableLayoutStyles.headerCell},
       },
-      {
-        property: 'notAnswered',
-        header: {
-          label: commonMsg.notAnswered(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-            width: 90,
-          }},
-        },
-        cell: {
-          format: notAnsweredFormatter,
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-            width: 90,
-          }}
-        }
+      cell: {
+        format: answerColumnsFormatter,
+        props: {style: tableLayoutStyles.cell},
+      }
+    }
+  );
+
+  getQuestionColumn = (sortable) => (
+    {
+      header: {
+        label: commonMsg.question(),
+        props: {style: tableLayoutStyles.headerCell},
+        transforms: [sortable],
       },
-    ];
+      cell: {
+        props: {style: tableLayoutStyles.cell},
+      }
+    }
+  );
+
+  getColumns = (sortable) => {
+    const maxAnswerChoicesLength = this.props.questionAnswerData.reduce((answersTotal, currentAnswerCount) => {
+      return Math.max(answersTotal, currentAnswerCount.answers.length);
+    }, 0);
+
+    let dataColumns = [];
+    let columns = this.getQuestionColumn(sortable) ;
+
+    dataColumns.push({property: 'question', ...columns});
```

Comment:
Can you add `property: 'question'` to the object returned by `getQuestionColumn()` so that you don't have to add it here? Then you can just say `dataColumns.push(this.getQuestionColumn(sortable))`.

Same for `property: 'percentAnswered'` and `property: 'notAnswered'` below.

Reply:
Fixed!